### PR TITLE
[typescript-fetch]: use ReponseError instead of throwing a response (#10477)

### PR DESCRIPTION
**Pull request by javier-garcia-meteologica**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@TiFu @taxpon @sebastianhaas  @kenisteward  @Vrolijkx @macjohnny @topce @akehir @petejohansonxo @amakhrov

### CODE REVIEW

**Comment by macjohnny** [comment]
Thanks for your contribution.
Since this is a breaking change, please file against the 6.0 branch.
@amakhrov what is your opinion here? Should we introduce an option?

**Comment by amakhrov** [comment]
> Should we introduce an option?

I don't think so. Migrating existing code should be trivial enough.

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache [hunk 1]
```diff
@@ -101,6 +101,13 @@ export class BaseAPI {
     }
 };
 
+export class ResponseError extends Error {
```
Comment (amakhrov) [inline modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache:104]: note that this comes with some gotchas and generally won't work if your TS compile target is `es5`.

e.g. see https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work

Comment (javier-garcia-meteologica) [reply modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache:104]: As a workaround, you can match by error name if your code needs to be compiled to es5

```ts
if (err.name === 'ResponseError') {
   ...
}
```

**Comment by mkusaka** [comment]
@macjohnny @amakhrov Is there any possibility this change will be included in version 6.0?
Currently, it seems that this pr is not included to 6.0 milestone.
Thanks.

**Comment by macjohnny** [comment]
@amakhrov are you fine with this change?

**Comment by amakhrov** [comment]
@macjohnny yes, it looks reasonable, and is definitely better than throwing an arbitrary response object

**Pull request merged**